File: SpendLess/SpendLess.Core/Models/Transaction.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace SpendLess.Core.Models
{
    public class Transaction
    {
        public string ImportHash { get; set; }
        public Guid AccountId { get; set; }
        public Guid Vendor { get; set; }
        public decimal Value { get; set; }
        public DateTime Date { get; set; }
        public bool Recurring { get; set; }
        public string Category { get; set; }
        public string Description { get; set; }
    }
}

File: Frontend/app/src/app/models/data/Transaction.ts
import { DecimalPipe } from "@angular/common";

export class TransactionModel {
    id: string;
    importHash?: string;
    vendor: string;
    value: number;
    date: string;
    recurring: boolean;
    category: string;
    description: string;

    _selected: boolean;

    static GetErrors(transaction: TransactionModel): string{
        let errors = [];
        if (!transaction.date){
            errors.push("date");
        }

        let errorMessage = "";
        if (errors.length > 0){
            errorMessage = "The following fields must be populated: ";
            errorMessage += errors.join(", ");
        }

        return errorMessage;
    }
}